.github/workflows/go.yml

```diff
@@ -300,3 +300,125 @@ jobs:
                   name: test-reports-modules
                   path: |
                       go/reports/**
+
+    test-pubsub:
```

Comment:
There is no need to have a completely separate job. PubSub tests are currently quick enough to run as part of the test-go job. For now, I would add it as just one more `make` task 
```yaml
            - name: Install & build & test
              working-directory: go
              run: |
                  make install-tools build
                  make -k unit-test integ-test pubsub-test
```